server/controllers/document.js

```diff
@@ -0,0 +1,113 @@
+const Document = require('../models/document');
+
+const DocumentCtrl = {
+  create(req, res) {
+    if (!req.body.title) {
+      return res.status(400).send({ error: 'Title required' });
+    }
+    if (!req.body.content) {
+      return res.status(400).send({ error: 'Content required' });
+    }
+    const doc = new Document();
+    doc.title = req.body.title;
+    doc.content = req.body.content;
+    if (req.body.public) { doc.public = req.body.public; }
+    doc.ownerId = req.decoded._doc._id;
+    if (req.body.role) { doc.role = req.body.role; } else { doc.role = req.decoded._doc.role; }
+
+    doc.save((err, document) => {
+      if (err) { return res.status(500).json({ error: err }); }
+      return res.status(201).send(document);
+    });
+  },
+  all(req, res) {
+    const limit = req.query.limit || 0;
+    const page = req.query.page || 0;
+    let endDate = Date.now();
+    let startDate;
+    const query = {};
+    startDate = new Date('1970-1-1');
+    if (req.query.date) {
+      startDate = new Date(req.query.date);
+      endDate = new Date(startDate.getTime() + (24 * 60 * 60 * 1000));
+    }
+
+    if (req.query.date) {
+      query.createdAt = {
+        $gte: startDate,
+        $lt: endDate,
+      };
+    }
+
+    if (req.decoded._doc.role === 'Users') {
+      query.role = 'Users';
+      query.public = true;
+    }
+
+    if (req.decoded._doc.role === 'Users') {
+      Document.find(query)
```

Comment:
**Code Quality - Critical**

Refactor code with the query object so that you reduce the number of code